dev/bots/count-github-tasks.py

```diff
@@ -0,0 +1,32 @@
+# Copyright 2019 The Chromium Authors. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+# Script that reports how many GitHub tasks autorollers should expect
+# on PRs on this repository.
+#
+# This assumes all the non-skipped tasks run, which is not accurate
+# for PRs that don't affect the engine version!
+
+# This file must be in Python because it is used by the Skia
+# autoroller logic and that logic does not have Dart available.
+
+import re
+
+# First count the Cirrus tasks...
+cirrusYaml = open('.cirrus.yml', 'r').readlines()
+count = 0;
+for line in cirrusYaml:
+    if re.search('^ +- name:', line):
```

Comment:
This isn't the only way to specify a task in Cirrus though, right?

What happens if/when someone lands a change to .cirrus.yml that doesn't use this format?

Reply:
One possibility: the auto-roller would catch the number mismatch, notify the engine rotation to change the script to capture the new format.

Reply:
How would it catch the nuymber mismatch though?

In Cirrus YAML file, you can define a task like this:

```yaml
foo_task:
  test_script: gradle test
```

Which implicitly gets the name from the top level declaration, or like this:

```yaml
task:
  - name: foo
  ...
```

Or like this:

```yaml
task:
  matrix:
    - name: fooA
    ...
```

At the very least, we need to count the number of top level `task` entries and then check if that entry actually is more than one task.  I think it would work to just have the counter increment by 1 for each `task:` line, and then increment it again for each `- name` line found before the next `task:` line (but decrement it by 1 if you find any `- name` lines) - but only until the format changes and/or adds yet another way to declare a task....

Reply:
@fkorotkov  - would it be difficult to add something to the Cirrus API to help with this?  Could we have some endpoint that tells us the total number of tasks that are expected to run for a given commit?  Or does such an endpoint exist?

Reply:
@dnfield : Skia's auto-roller can tell if the Github API returns a number of tests that's greater than what this script returns. If everything is up to date, Github API should only return a number that's no greater than the one returned by this script. See https://github.com/flutter/flutter/issues/43672#issuecomment-548504586

Reply:
But the GitHub API might return a number that exactly matches this count, and this count is too low, so everything looks good but is not, right?

Reply:
We are trying to keep our cirrus.yml file sane, which includes being extremely consistent about how we declare tasks. So we should never use the alternatives described above.

Reply:
Could this script enforce that then, such that we check that no line matches `.+_task:`?

Reply:
@dnfield : right, we could get false positives from Github API. I guess the chance that we'll get an alert while there's a mismatch should be quite high (say > 50%). Let's see how it goes, and I think it's at least better than our current state where there's 0% chance of knowing a mismatch automatically.

Reply:
yeah, that kind of test seems like it would make sense